components/contributors/InfoCard.tsx

```diff
@@ -5,16 +5,31 @@ import Image from "next/image";
 import Link from "next/link";
 import { BsSlack } from "react-icons/bs";
 import { env } from "@/env.mjs";
+import { getLeaderboardData } from "@/app/api/leaderboard/functions";
+import { parseDateRangeSearchParam } from "@/lib/utils";
 
-export default function InfoCard({
+export default async function InfoCard({
   contributor,
+  isRank,
   minimal = true,
   isClickable = false,
 }: {
   contributor: Contributor;
+  isRank?: boolean;
   minimal?: boolean;
   isClickable?: boolean;
 }) {
+  const leaderboardData = await getLeaderboardData(
+    parseDateRangeSearchParam(),
+    "points",
+    "desc",
+    [],
+  );
+  const userPosition = leaderboardData.findIndex(
+    (data) => data.user.social.github === contributor.github,
+  );
+  const hideBadges = userPosition === -1;
```

Comment:
It's fine to inline this in the JSX itself. Also even though you are naming it as `hideBadges` this variable seems to have nothing to do with the badges.